### ISSUE TITLE
Ability to work with php on WSL (Ubuntu on Bash on Windows)

### ISSUE THREAD

**Issue by erdemece** [felixfbecker/vscode-php-intellisense#272]
I have added bellow bat script to my windows environment and I am able to use php command from cmd (on windows environment) 

But this extension still asking for php.exe which I don't want to use. Can you please fix this? 

Thanks.

`@echo OFF
setlocal ENABLEDELAYEDEXPANSION

rem Collect the arguments and replace:
rem  '\' with '/'
rem  'c:' with 'mnt/c'
rem  '"' with '\"'
set v_params=%*
set v_params=%v_params:\=/%
set v_params=%v_params:C:=/mnt/c%
set v_params=%v_params%
set v_params=%v_params:"=\"%

rem Call the windows-php inside WSL.
rem windows-php is just a script which passes the arguments onto
rem the original php executable and converts its output from UNIX
rem syntax to Windows syntax.
C:\Windows\sysnative\bash.exe -l -c "windows-php %v_params%"`

**Comment by felixfbecker** [comment]
Seems like you figured out a solution, what exactly do you want to be "fixed"?
Also what speaks against PHP on Windows?

**Comment by erdemece** [comment]
@felixfbecker  thanks for response.

even tough I set the php path setting as this "php.executablePath": "C:\\Users\\erdem\\batfiles\\php.bat", everytime I open vscode I get following error "Cannot validate since no PHP executable is set 'php.validate.executablePath' to configure the PHP executable." 

I am not sure if the extension works at this point.

I have disabled php.suggest.basic as well. PHP suggests me basic commands.

**Comment by erdemece** [comment]
@felixfbecker sorry to be clear

I have changed my bat code to this

`@echo off
set v_params=%*
set v_params=%v_params:\=/%
set v_params=%v_params:c:=/mnt/c%
set v_params=%v_params:"=\"%
@bash -c "php %v_params%"`

if I change the settings as below I don't get error but vscode doesn't suggest any php code;

    "php.validate.executablePath": "E:\\paths\\php.bat",
    "php.executablePath": "E:\\paths\\php.bat",

If I change this to;

    "php.validate.executablePath": "E:\\paths\\php.bat",
    "php.executablePath": "E:\\paths\\php7\\php.exe",

It works. My point is I would like to be able to use same php version.

Thanks.

**Comment by felixfbecker** [comment]
It would be nice if you could explain the reason why you can’t use a Windows binary, otherwise I don’t really understand why this feature needs to be added

**Comment by erdemece** [comment]
Well, I would like to use same server or php that I test my code via browser and use same php with my code editor.

**Comment by MadHed** [comment]
I don't think this is going to work. Even if you manage to start the php executable in WSL from vscode it would see a completely different path than vscode and the directory<->URI mapping will be messed up.

**Comment by sagarrabadiya** [comment]
this will be much nicer to have.
